rotkehlchen/chain/ethereum/modules/convex/decoder.py

```diff
@@ -0,0 +1,107 @@
+from rotkehlchen.chain.ethereum.types import string_to_ethereum_address
+from rotkehlchen.types import ChecksumEthAddress, EthereumTransaction
+
+CVX_REWARDS = string_to_ethereum_address('0xCF50b810E57Ac33B91dCF525C6ddd9881B139332')
+CVXCRV_REWARDS = string_to_ethereum_address('0x3Fe65692bfCD0e6CF84cB1E7d24108E434A7587e')
+BOOSTER = string_to_ethereum_address('0xF403C135812408BFbE8713b5A23a04b3D48AAE31')
+
+CLAIMZAP_OLD = string_to_ethereum_address('0x92Cf9E5e4D1Dfbf7dA0d2BB3e884a68416a65070')
+CLAIMZAP_NEW = string_to_ethereum_address('0xDd49A93FDcae579AE50B4b9923325e9e335ec82B')
+
+CVX_WITHDRAW_TOPIC = b'p\x84\xf5Gf\x18\xd8\xe6\x0b\x11\xef\r}?\x06\x91FU\xad\xb8y>(\xff\x7f\x01\x8dLv\xd5\x05\xd5'  # noqa: E501
+CVXCRV_REWARD_PAID_TOPIC = b'\xe2@6@\xbah\xfe\xd3\xa2\xf8\x8buWU\x1d\x19\x93\xf8K\x99\xbb\x10\xff\x83?\x0c\xf8\xdb\x0c^\x04\x86'  # noqa: E501
+CVXCRV_WITHDRAWAL_TOPIC = b'p\x84\xf5Gf\x18\xd8\xe6\x0b\x11\xef\r}?\x06\x91FU\xad\xb8y>(\xff\x7f\x01\x8dLv\xd5\x05\xd5'  # noqa: E501
+
+
+class ConvexDecoder(DecoderInterface):
+    @staticmethod
+    def _decode_cvxcrv_rewards(
+            tx_log: EthereumTxReceiptLog,
+            transaction: EthereumTransaction,  # pylint: disable=unused-argument
+            decoded_events: List[HistoryBaseEntry],
+            all_logs: List[EthereumTxReceiptLog],  # pylint: disable=unused-argument
+            action_items: List[ActionItem],  # pylint: disable=unused-argument
+
+    ) -> Tuple[Optional[HistoryBaseEntry], Optional[ActionItem]]:
+        for event in decoded_events:
+            if event.event_type == HistoryEventType.RECEIVE and event.event_subtype == HistoryEventSubType.NONE:  # noqa: E501
+                if tx_log.topics[0] == CVXCRV_REWARD_PAID_TOPIC:
+                    event.event_subtype = HistoryEventSubType.REWARD
+                elif tx_log.topics[0] == CVXCRV_WITHDRAWAL_TOPIC:
+                    event.event_type = HistoryEventType.WITHDRAWAL
+                event.counterparty = CPT_CONVEX
+            elif event.event_type == HistoryEventType.SPEND and event.event_subtype == HistoryEventSubType.NONE:  # noqa: E501
+                event.event_type = HistoryEventType.DEPOSIT
+                event.counterparty = CPT_CONVEX
+
+        return None, None
+
+    @staticmethod
+    def _decode_booster(
+            tx_log: EthereumTxReceiptLog,  # pylint: disable=unused-argument
+            transaction: EthereumTransaction,  # pylint: disable=unused-argument
+            decoded_events: List[HistoryBaseEntry],
+            all_logs: List[EthereumTxReceiptLog],  # pylint: disable=unused-argument
+            action_items: List[ActionItem],  # pylint: disable=unused-argument
+
+    ) -> Tuple[Optional[HistoryBaseEntry], Optional[ActionItem]]:
+        for event in decoded_events:
+            if event.event_type == HistoryEventType.SPEND and event.event_subtype == HistoryEventSubType.NONE:  # noqa: E501
+                if event.counterparty == ZERO_ADDRESS:
```

Comment:
Do the events have any special tx topic to identify them?

Reply:
I could check it again, but as far as I remember, they don't